### PR TITLE
feat(event): add yaml for stats

### PR DESCRIPTION
**Pull request by sinhaashish**

### CODE REVIEW

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors try

File: chart/templates/mayastor/obs/stats-configmap.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-configmap.yaml:11]: must we include the stats here? Otherwise we need to update when the schema changes and IIRC we didn't want to have to manage changes etc

File: chart/templates/mayastor/rbac/rbac.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/rbac/rbac.yaml:38]: don't think we need all of these?

File: chart/values.yaml
Comment (tiagolobocastro) [inline chart/values.yaml:618]: this should be within eventing

**Comment by tiagolobocastro** [comment]
I don't think this can be merged until 253 is ready? Otherwise images will not be available and deployment will fail which may fail e2e? CC @blaisedias

**Comment by sinhaashish** [comment]
> I don't think this can be merged until 253 is ready? Otherwise images will not be available and deployment will fail which may fail e2e? CC @blaisedias

Ok , in that case i will have this a s a part of 253 only.

**Comment by tiagolobocastro** [comment]
> > I don't think this can be merged until 253 is ready? Otherwise images will not be available and deployment will fail which may fail e2e? CC @blaisedias
> 
> Ok , in that case i will have this a s a part of 253 only.

Can be separate, just don't merge this until #253 is merged :)

File: chart/values.yaml
Comment (sinhaashish) [reply chart/values.yaml]: done

File: chart/templates/mayastor/rbac/rbac.yaml
Comment (sinhaashish) [reply chart/templates/mayastor/rbac/rbac.yaml:41]: needed as we need to get the release name form api rest post . release name is the prefix for config map.

File: chart/values.yaml
Comment (tiagolobocastro) [inline chart/values.yaml:634]: was this intentional?

**Review by tiagolobocastro — APPROVED** [review]

File: chart/values.yaml
Comment (sinhaashish) [reply chart/values.yaml]: removed

File: chart/templates/mayastor/obs/stats-configmap.yaml
Comment (Abhinandan-Purkait) [inline chart/templates/mayastor/obs/stats-configmap.yaml:10]: Is this going to be populated by the components?

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (Abhinandan-Purkait) [inline chart/templates/mayastor/obs/stats-service.yaml:19]: Don't we need a customisable type? Is it always going to be ClusterIP?

ref:- https://github.com/openebs/mayastor-extensions/blob/develop/chart/templates/mayastor/apis/api-rest-service.yaml

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (datacore-vvarakantham) [inline chart/templates/mayastor/obs/stats-service.yaml:13]: I think the port should be 9090. It was decided that 9090 is to be used to expose the stats.

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (datacore-vvarakantham) [inline chart/templates/mayastor/obs/stats-service.yaml:5]: Can we use just stats instead of stats aggregator?

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (sinhaashish) [reply chart/templates/mayastor/obs/stats-deployment.yaml:33]: Its correct, the stats binary is run  here.

File: chart/templates/mayastor/obs/stats-configmap.yaml
Comment (sinhaashish) [reply chart/templates/mayastor/obs/stats-configmap.yaml:10]: The data field is going to be populated from code . If we have the value here then we need to update when the schema changes. So no data here .

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (sinhaashish) [reply chart/templates/mayastor/obs/stats-service.yaml:19]: As far as i can recall , it was stated that we use cluster ip only.

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (sinhaashish) [reply chart/templates/mayastor/obs/stats-service.yaml:5]: done

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (niladrih) [reply chart/templates/mayastor/obs/stats-service.yaml:19]: It should be customizable IMO.

**Review by niladrih — COMMENTED** [review]
Left few comments

**Comment by niladrih** [comment]
I think there is some naming convention for the components' metadata.names. I don't recall at the moment. Maybe @tiagolobocastro can help with this.

**Comment by tiagolobocastro** [comment]
> I think there is some naming convention for the components' metadata.names. I don't recall at the moment. Maybe @tiagolobocastro can help with this.

Good shout, have we named components as per convention here? @sinhaashish @avishnu

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-service.yaml:15]: Do we need to add custom portType here?

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (tiagolobocastro) [reply chart/templates/mayastor/obs/stats-deployment.yaml:33]: Don't specify the command, only the args are needed

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/obs/stats-service.yaml]: I think should be customisable with cluster ip as default. But I will leave upto you.

**Review by Abhinandan-Purkait — APPROVED** [review]

File: chart/templates/mayastor/obs/stats-service.yaml
Comment (sinhaashish) [reply chart/templates/mayastor/obs/stats-service.yaml]: Addressed your comment.

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-deployment.yaml:5]:
```suggestion
  name: {{ .Release.Name }}-obs-callhome-stats
```
?

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-deployment.yaml:14]:
```suggestion
      app: obs-callhome-stats
```
also on the other ones

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-deployment.yaml:30]:
```suggestion
        - name: obs-callhome-stats
```

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-deployment.yaml:33]:
```suggestion
```

File: chart/values.yaml
Comment (tiagolobocastro) [inline chart/values.yaml:594]: sorry since we've discussed that stats are callhome specific now I think this should actually be inside callhome and not eventing.

Comment (sinhaashish) [reply chart/values.yaml]: done. PTAL @tiagolobocastro

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors try

File: chart/templates/mayastor/rbac/rbac.yaml
Comment (niladrih) [inline chart/templates/mayastor/rbac/rbac.yaml:38]: Now that we are creating the configmap using code.
```suggestion
  verbs: ["create", "get", "update", "patch"]
```

File: chart/values.yaml
Comment (niladrih) [inline chart/values.yaml:636]: This comment should mention how to disable stats

**Review by niladrih — COMMENTED** [review]
Few minor comments..

File: chart/values.yaml
Comment (niladrih) [reply chart/values.yaml:636]: We can mention that in the comment.

Comment (sinhaashish) [reply chart/values.yaml]: done

File: chart/Chart.yaml
Comment (tiagolobocastro) [inline chart/Chart.yaml:42]: this should be eventing still right?

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-deployment.yaml:1]: this should be callhome and eventing enabled?

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (tiagolobocastro) [inline chart/templates/mayastor/obs/stats-deployment.yaml:3]: Btw, can't we just add this as a separate container within the callhome deployment, is there any benefit to having this separate?

**Review by niladrih — APPROVED** [review]
lgtm

**Comment by sinhaashish** [comment]
bors merge

File: chart/templates/mayastor/obs/stats-deployment.yaml
Comment (sinhaashish) [reply chart/templates/mayastor/obs/stats-deployment.yaml:3]: addressed in PR #253

File: chart/Chart.yaml
Comment (sinhaashish) [reply chart/Chart.yaml:42]: addressed in PR #253

**Pull request merged**
